File: Python/Bubble-Sort.py
from random import randint


def is_sorted(arr):
    for x in range(len(arr)-2):
        if arr[x] > arr[x+1]:
            return False

    return True


def bubble_sort(arr):
    done = False
    length = len(arr)

    while not done:
        done = True
        for x in range(length-1):
            if arr[x] > arr[x+1]:
                arr[x], arr[x+1] = arr[x+1], arr[x]
                done = False

    return arr


if __name__ == '__main__':
    arr = [randint(1, 100) for _ in range(10)]
    print(f"Array: {arr}. Sorted: {is_sorted(arr)}")
    arr = bubble_sort(arr)
    print(f"Array: {arr}. Sorted: {is_sorted(arr)}")

File: C/Bubble-Sort.c
#include <stdio.h>
#include <stdbool.h>


bool is_sorted(int *arr, int len) {
    for(int i=0; i<len-1; ++i) {
        if(arr[i] > arr[i+1])
            return false;
    }

    return true;
}


void print(int *arr, int len) {
    printf("Array: ");
    for(int i=0; i<len; ++i)
        printf("%d ", arr[i]);
    printf(". Sorted? %s\n", (is_sorted(arr, len)?"Yes": "No"));
}


void bubble_sort(int *arr, int len) {
    bool done = false;

    while(!done) {
        done = true;

        for(int i=0; i<len-1; ++i) {
            if(arr[i] > arr[i+1]) {
                int temp = arr[i];
                arr[i] = arr[i+1];
                arr[i+1] = temp;
                done = false;
            }
        }
    }
}


int main() {
    int arr[] = {5, 3, 4, 7, 2, 1};
    int len = 6;
    print(arr, len);
    bubble_sort(arr, len);
    print(arr, len);

    return 0;
}
